tests/cephadm/test_os_tuning_profile.py

```diff
@@ -0,0 +1,94 @@
+import tempfile
+
+import yaml
+
+from cli.cephadm.cephadm import CephAdm
+from utility.log import Log
+
+log = Log(__name__)
+
+
+class OsTuningProfileError(Exception):
+    pass
+
+
+def generate_tuned_profile_spec(installler_node, specs):
+    """
+    Generate tuned profile.
+    Args:
+        installler_node: installer node
+        specs: tuned profile specs
+    """
+
+    log.info(f"Spec yaml file content:\n{specs}")
+    # Create tuned profile spec yaml file
+    temp_file = tempfile.NamedTemporaryFile(suffix=".yaml")
+    spec_file = installler_node.remote_file(
+        sudo=True, file_name=temp_file.name, file_mode="wb"
+    )
+    spec = yaml.dump(specs, sort_keys=False, indent=2).encode("utf-8")
+    spec_file.write(spec)
+    spec_file.flush()
+    return temp_file.name
+
+
+def run(ceph_cluster, **kw):
+    """Verify tuning profile test cases.
+    Args:
+        **kw: Key/value pairs of configuration information to be used in the test.
+        kw: test data
+        e.g:
+        test:
+            name: create_tuning_profile
+            desc: Verify tuning profile created successfully using cephadm
+            polarion-id: CEPH-83575318
+            module: test_os_tuning_profile.py
+            config:
+                command: apply
+                specs:
+                    profile_name: test-mon-host-profile
+                    placement:
+                        hosts:
+                            - node1
+                            - node2
+                settings:
+                    fs.file-max: 100000
+                    vm.swappiness: 14
+                apply_result: Saved tuned profile test-mon-host-profile
+                ls_result: "profile_name: test-mon-host-profile"
+    """
+    installler_node = ceph_cluster.get_nodes(role="installer")[0]
+    config = kw["config"]
+    mount = "/tmp:/tmp"
+    if config["command"] == "apply" or config["command"] == "re-apply":
+        nodes = config.get("specs", {}).get("placement", {}).get("hosts")
+        hosts = [ceph_cluster.get_nodes()[int(node[-1])].hostname for node in nodes]
+        config["specs"]["placement"]["hosts"] = hosts
+        specs = config["specs"]
```

Comment:
`specs` variable is not resued anywhere. Below function call can be modified as -
```
spec_file = generate_tuned_profile_spec(installler_node, config.get("specs"))
```

Reply:
Updated